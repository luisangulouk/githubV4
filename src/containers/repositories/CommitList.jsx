/* eslint-disable no-console */
import React from 'react';
import Style from './repositories.scss';

const CommitList = ({
  history
}) => {
  return (
    <ul className={Style.repoList}>
      {history.edges.map(({node}) => {
        console.log(node);
        return (
          <li key={node.oid} className={Style.branchCommitMessage}>
            <span>{node.messageHeadline}</span>
          </li>
        );
      }
      )}
    </ul>
  );
};

export default CommitList;
