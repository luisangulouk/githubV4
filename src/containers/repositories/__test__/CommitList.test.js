import React from 'react';
import renderer from 'react-test-renderer';

import CommitList from '../CommitList';

describe('Commit List Component', () => {

  const mockHistory = {
    edges: [{
      node: {
        oid: 'fd91ff7157613b03eb33388761f5f3a769d4091c',
        messageHeadline: '[GHV4] refactor: Add handleClick function on commits, add test',
        __typename: 'Commit'
      },
      __typename: 'CommitEdge'
    }
    ],
    pageInfo: {
      hasNextPage: true,
      endCursor: 'fd91ff7157613b03eb33388761f5f3a769d4091c 9',
      __typename: 'PageInfo'
    },
    __typename: 'CommitHistoryConnection'
  };

  it('should match snapshot', () => {

    const component = renderer.create(<CommitList history={mockHistory} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
