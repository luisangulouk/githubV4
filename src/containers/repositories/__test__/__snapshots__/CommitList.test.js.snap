// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Commit List Component should match snapshot 1`] = `
<ul
  className="repoList"
>
  <li
    className="branchCommitMessage"
  >
    <span>
      [GHV4] refactor: Add handleClick function on commits, add test
    </span>
  </li>
</ul>
`;
